### PR TITLE
fix: Make tests account agnostic

### PR DESCRIPTION
**Pull request by kbanman**
Wanted to be able to run the tests on my machine, and had to make some changes to make that happen.

Also corrected some spelling and added a simple `testing.md` to document my understanding of the setup needed to run e2e tests.

### CODE REVIEW

**Comment by Lokicoule** [comment]
Hello, you can find  [here](https://github.com/Lokicoule/nestjs-cognito/wiki) all the informations needed.

File: testing.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Testing
```
Comment (Lokicoule) [inline testing.md:1]: Must be split into an other commit!

File: packages/testing/tests/app.module.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { CognitoTestingModule } from "../lib";
   imports: [
     ConfigModule.forRoot(),
     CognitoTestingModule.register({
-      region: "eu-west-1",
+      region: process.env.COGNITO_REGION ?? "eu-west-1",
```
Comment (Lokicoule) [inline packages/testing/tests/app.module.ts:9]: Use configService instead:
```ts
CognitoTestingModule.registerAsync({
      imports: [ConfigModule.forRoot()],
      useFactory: async (configService: ConfigService) => ({
        region: configService.get("COGNITO_REGION"),
      }),
      inject: [ConfigService],
    }),
```

File: testing.md (hunk 1 above)
Comment (Lokicoule) [reply testing.md:1]: The workflow is well documented and helps to shed some light on the wiki's shadows.
Thank you!

File: .gitignore [hunk 3]
```diff
@@ -34,6 +34,7 @@ yarn.lock
 *.launch
 .settings/
 *.sublime-workspace
+.vscode
```
Comment (Lokicoule) [inline .gitignore:37]: Merge line 37 with line 40 into a single regular expression.

**Review by Lokicoule — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (kbanman) [reply .gitignore]: I actually didn't notice the existing vscode section before 🤦🏻‍♂️. Removed this line entirely.

File: packages/testing/tests/app.module.ts (hunk 2 above)
Comment (kbanman) [reply packages/testing/tests/app.module.ts]: Right, should have caught that

File: testing.md (hunk 1 above)
Comment (kbanman) [reply testing.md]: My mistake, I didn't see the wiki at all! Will remove this file and instead update the wiki.

**Comment by kbanman** [comment]
Hey Loïk, looks like the CI tests are failing due to a missing `Region` parameter. Were those passing prior to my changes?

The e2e tests are working on my machine, so I expect it would be an environment thing, maybe a missing env var?

**Comment by Lokicoule** [comment]
Hi Kelly,

I manually cherry-picked your commits and pushed them to the main repository, everything works fine although it's not a perfect git workflow.
I guess the CI tests failed because you haven't set it up on your side and you need it when it's a fork repository (to confirm).

If you're ok with the git fix, we can close the PR.

File: testing.md (hunk 1 above)
Comment (Lokicoule) [reply testing.md]: I updated the [wiki](https://github.com/Lokicoule/nestjs-cognito/wiki) with the contents of your testing file.
Thank you kelly ;)

**Comment by kbanman** [comment]
Thanks!

**Pull request closed without merging**